### DISCUSSION TITLE
Is there a tool to generate a stub of the docs/ section of a component?

### DISCUSSION THREAD

**Discussion by argargreg** [nasa/fprime#1903] [Q&A] [answered]
Dear developers,

Many built-in F' components follow the same template for their docs/sdd.md file, including a table + image description of the component. The image is stored in the img subfolder of docs. 

It feels like it was probably worth it for the developers to create a tool to auto-generate the docs/ folder, and let the component's author fill in the relevant sections. However I could not find anything anywhere that could confirm or invalidate that.

Note that I am not referring to the tool to create a graphical representation of a topology (fprime-layout). I am hoping for a similar tool that accepts fpp files and not topology XML files (which would already be a win for me), part of another tool that would create the sdd.md file following a set template and generate the Ports section content.

Does such a tool exist? If so, how could we use it?

Thank you in advance.

**Comment by LeStarch** [comment, marked as answer]
This tool does not yet exist. We do intend to create it someday.

**Reply by SterlingPeet** [reply]
My ``fprime-extras`` project includes a ``docs`` command to generate template docs.  It is woefully out of date, requires F Prime in version 1.X or maybe 2.X, but it did create all of the relevant markdown documentation from the XML files in use at the time.  It does not know anything about FPP and uses the older documentation format, so its an overhaul to update it.

However, such a thing has existed before.

**Reply by SterlingPeet** [reply]
The reason it did not get updated in-house is that we use a custom cookiecutter template that generates the bare-bones markdown docs, and when a developer fails to fill them out we catch it at the component PR review.

**Reply by argargreg** [reply]
Good to know, thanks to both.
